kubernetes/deployment-production.tmpl

```diff
@@ -0,0 +1,300 @@
+          image: zooniverse/talk-api:__IMAGE_TAG__
+          resources:
+            requests:
+              memory: "700Mi"
+              cpu: "50m"
+              ephemeral-storage: "1Gi"
+            limits:
+              memory: "700Mi"
+              cpu: "500m"
+              ephemeral-storage: "1Gi"
+          livenessProbe:
+            exec:
+              command:
+                - /rails_app/docker/sidekiq_status
+            initialDelaySeconds: 20
+          args: ["/rails_app/scripts/docker/start-sidekiq.sh"]
+          envFrom:
+          - secretRef:
+              name: talk-common-env-vars
+          - secretRef:
+              name: talk-production-env-vars
+          - configMapRef:
+              name: talk-staging-shared
+          volumeMounts:
+          - mountPath: /tmp
+            name: talk-production-dumpworker-data
+      volumes:
+        - name: talk-production-dumpworker-data
+          emptyDir: {}
+---
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: talk-production-redis
+spec:
+  accessModes:
+    - ReadWriteOnce
+  storageClassName: azurefile
+  resources:
+    requests:
+      storage: 5Gi
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: talk-production-redis
+  labels:
+    app: talk-production-redis
+spec:
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      app: talk-production-redis
+  template:
+    metadata:
+      labels:
+        app: talk-production-redis
+    spec:
+      tolerations:
+      - key: "servicelife"
+        operator: "Equal"
+        value: "longlife"
+        effect: "NoSchedule"
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: servicelife
+                operator: In
+                values:
+                - longlife
+      initContainers:
+        - name: disable-thp
+          image: busybox
+          volumeMounts:
+            - name: host-sys
+              mountPath: /host-sys
+          command: ["sh", "-c", "echo never >/host-sys/kernel/mm/transparent_hugepage/enabled"]
+      containers:
+        - name: talk-production-redis
+          image: redis:6.0.5
+          resources:
+            requests:
+              memory: "2500Mi"
+              cpu: "500m"
+            limits:
+              memory: "2500Mi"
+              cpu: "1500m"
+          volumeMounts:
+          - name: talk-production-redis-data
+            mountPath: "/data"
+      volumes:
+        - name: talk-production-redis-data
+          persistentVolumeClaim:
+            claimName: talk-production-redis
+        - name: host-sys
+          hostPath:
+            path: /sys
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: talk-production-redis
+spec:
+  selector:
+    app: talk-production-redis
+  ports:
+    - protocol: TCP
+      port: 6379
+      targetPort: 6379
+  type: NodePort
+---
```

Comment:
We can now use DNS-based validation with Let's Encrypt, so we should manually define the certificate like this to get the certificate issued before we cut over the live traffic (we need to manually define it to add the `use-azuredns-solver` label):
```suggestion
---
apiVersion: certmanager.k8s.io/v1alpha1
kind: Certificate
metadata:
  name: talk-production-tls
  labels:
    use-azuredns-solver: "true"
spec:
  issuerRef:
    name: letsencrypt-prod
    kind: ClusterIssuer 
  secretName: talk-production-tls
  dnsNames:  
    - talk.zooniverse.org
    - talk.azure.zooniverse.org
---
```

Reply:
Cool, added. Gonna ask more about this, since nothing else seems to do this.